### PR TITLE
Add register custom partial functionality

### PR DESCRIPTION
**Pull request by nmanu1**
Allow custom Handlebars partials to be registered, similar to templates and helpers.

J=SLAP-1544
TEST=manual

Local testing by adding and using a partial in theme

### CODE REVIEW

File: src/ui/rendering/handlebarsrenderer.js [hunk 1]
```diff
@@ -51,6 +51,15 @@ export default class HandlebarsRenderer extends Renderer {
     this._handlebars.registerHelper(name, cb);
   }
 
+  /**
+   * register a partial to the internal Handlebars compiler
+   * @param {*} partialName The unique name for the partial
```
Comment (tmeyer2115) [inline src/ui/rendering/handlebarsrenderer.js]: Update the param types here

File: src/ui/rendering/handlebarsrenderer.js [hunk 2]
```diff
@@ -51,6 +51,15 @@ export default class HandlebarsRenderer extends Renderer {
     this._handlebars.registerHelper(name, cb);
   }
 
+  /**
+   * register a partial to the internal Handlebars compiler
```
Comment (tmeyer2115) [inline src/ui/rendering/handlebarsrenderer.js]: Capitalize register

**Review by tmeyer2115 — COMMENTED** [review]
We should add a blurb about this in the README.

File: README.md [hunk 3]
```diff
@@ -2319,6 +2319,41 @@ ANSWERS.addComponent('MyCustomComponent', {
 });
 ```
 
+# Custom Partials
+
+[Handlebars partials](https://handlebarsjs.com/guide/partials.html) are
+templates that can be reused and called directly from other templates. The 
+SDK offers an `ANSWERS.registerPartial` function. This will map a partial string
```
Comment (cea2aj) [inline README.md]: nit: Since this repo is no longer officially called the SDK, I would reword this to "The ANSWERS object offers an `ANSWERS.registerPartial` function for custom partials."

File: README.md [hunk 4]
```diff
@@ -2319,6 +2319,41 @@ ANSWERS.addComponent('MyCustomComponent', {
 });
 ```
 
+# Custom Partials
+
+[Handlebars partials](https://handlebarsjs.com/guide/partials.html) are
+templates that can be reused and called directly from other templates. The
```
Comment (tmeyer2115) [inline README.md]: We need to be careful with the verbiage here. Templates have a very specific meaning in Handlebars and are distinct from Partials. A partial can be called from a template or another partial.

File: README.md [hunk 5]
```diff
@@ -2319,6 +2319,41 @@ ANSWERS.addComponent('MyCustomComponent', {
 });
 ```
 
+# Custom Partials
```
Comment (cea2aj) [inline README.md]: Can we add this to the README table of contents? It's located at the top of the README.

Since this section is similar to the "Template Helpers" section, what do you think of making an H1 section called "Custom Rendering", and inside that we have H2 sections for "Custom Partials" and "Template Helpers"?

Comment (nmanu1) [reply README.md]: Yeah, I was also thinking it might be nice to group them. Would the H1 title be confusing since there is also a "Using a Custom Renderer" section that might look similar?

Comment (cea2aj) [reply README.md]: Maybe we have the H1 be "Extending the built-in renderer"? That might be distinct enough from "Using a custom renderer"

File: README.md (hunk 4 above)
Comment (nmanu1) [reply README.md]: Good to know! I've reworded this a little in the latest commit, but let me know if I should change it more

**Review by tmeyer2115 — APPROVED** [review]
